web-common/src/components/column-profile/columns-profile-data.ts

```diff
@@ -0,0 +1,261 @@
+import {
+  loadColumnCardinality,
+  loadColumnsNullCount,
+  loadColumnTopK,
+  loadTableCardinality,
+} from "@rilldata/web-common/components/column-profile/column-types/common-data";
+import {
+  loadColumnHistogram,
+  loadDescriptiveStatistics,
+} from "@rilldata/web-common/components/column-profile/column-types/numeric-profile-data";
+import { loadTimeSeries } from "@rilldata/web-common/components/column-profile/column-types/timestamp-profile-data";
+import { getMapFromArray } from "@rilldata/web-common/lib/arrayUtils";
+import { createThrottler } from "@rilldata/web-common/lib/create-throttler";
+import {
+  CATEGORICALS,
+  INTERVALS,
+  isFloat,
+  isNested,
+  NUMERICS,
+  TIMESTAMPS,
+} from "@rilldata/web-common/lib/duckdb-data-types";
+import type {
+  NumericHistogramBinsBin,
+  NumericOutliersOutlier,
+  TopKEntry,
+  V1NumericStatistics,
+  V1TableColumnsResponse,
+  V1TimeGrain,
+  V1TimeSeriesValue,
+} from "@rilldata/web-common/runtime-client";
+import { BatchedRequest } from "@rilldata/web-common/runtime-client/batched-request";
+import { waitUntil } from "@rilldata/web-local/lib/util/waitUtils";
+import type { QueryObserverResult } from "@tanstack/query-core";
+import { getContext, setContext } from "svelte";
+import { Updater, writable } from "svelte/store";
+import type { Readable } from "svelte/store";
+
+export type ColumnProfileData = {
+  name: string;
+  type: string;
+
+  isFetching: boolean;
+  nullCount?: number;
+  cardinality?: number;
+
+  topK?: Array<TopKEntry>;
+
+  // numeric profile
+  rugHistogram?: Array<NumericOutliersOutlier>;
+  histogram?: Array<NumericHistogramBinsBin>;
+  descriptiveStatistics?: V1NumericStatistics;
+
+  // timestamp profile
+  estimatedRollupInterval?: V1TimeGrain;
+  smallestTimeGrain?: V1TimeGrain;
+  timeSeriesData?: Array<V1TimeSeriesValue>;
+  timeSeriesSpark?: Array<V1TimeSeriesValue>;
+};
+export type ColumnsProfileData = {
+  isFetching: boolean;
+  tableRows: number;
+  columnNames: Array<string>;
+  profiles: Record<string, ColumnProfileData>;
+};
+export type ColumnsProfileDataMethods = {
+  load: (
+    instanceId: string,
+    tableName: string,
+    profileColumnResponse: QueryObserverResult<V1TableColumnsResponse>
+  ) => Promise<void>;
+};
+export type ColumnsProfileDataStore = Readable<ColumnsProfileData> &
+  ColumnsProfileDataMethods;
```

Comment:
All this data/state looks like server-side data, which should be managed by TanStack Query. Can we use the TanStack stores rather than syncing the data into our own store?

Reply:
We would still need derived stores, check the files that were deleted. This is a simpler approach than that. Besides it is simpler for the fetcher code to update these stores.